src/components/Ingredients/IngredientForm.tsx

```diff
@@ -0,0 +1,132 @@
+import React, { useState } from 'react';
+import { useDispatch } from 'react-redux';
+
+import { CONFIRM_ICON, TRASH_ICON } from '../../helpers/icons';
+import { editIngredient, addIngredient } from '../../store/recipes/actions';
+import { v4 as uuidv4 } from 'uuid';
+import { IIngredient } from '../../store/interfaces';
+
+const IngredientForm = ({
+  recipeId,
+  ingredientId,
+  amount,
+  name,
+  unit,
+  isIngredientInEdition, //false=empty; true=filled
+  isNewIngredientAdded,
+  closeIngredientEdition,
+  deleteCurrentIngredient,
+  deleteNewIngredient,
+}: {
+  recipeId: number | string;
+  ingredientId: number | string;
+  amount: number;
+  name: string;
+  unit: string;
+  isIngredientInEdition?: boolean; //false=empty; true=filled
+  isNewIngredientAdded: boolean;
+  closeIngredientEdition?: () => void;
+  deleteCurrentIngredient: (ingredientId: number | string) => void;
+  deleteNewIngredient: () => void;
+}) => {
+  const [amountInput, setAmountInput] = useState<number>(amount);
```

Comment:
Co do inferencji to np. właśnie w tym miejscu, gdyby nie zatypować `<number>` explicit, to TS i tak wyznaczyłby sobie ten typ na podstawie inicjalizacji z wartością `amount`, która jest już otypowana w propsach (`amount: number`).